pkg/providers/instance/instance.go

```diff
@@ -18,57 +18,153 @@ package instance
 
 import (
 	"context"
+	"fmt"
 
+	ecsclient "github.com/alibabacloud-go/ecs-20140526/v4/client"
+	util "github.com/alibabacloud-go/tea-utils/v2/service"
+	"github.com/alibabacloud-go/tea/tea"
+	"go.uber.org/multierr"
 	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
 	"sigs.k8s.io/karpenter/pkg/cloudprovider"
 
 	"github.com/cloudpilot-ai/karpenter-provider-alicloud/pkg/apis/v1alpha1"
+	"github.com/cloudpilot-ai/karpenter-provider-alicloud/pkg/operator/options"
+	"github.com/cloudpilot-ai/karpenter-provider-alicloud/pkg/utils/alierrors"
 )
 
 type Provider interface {
-	Create(context.Context, *v1alpha1.ECSNodeClass, *karpv1.NodeClaim, []*cloudprovider.InstanceType) (*Instance, error)
-	Get(context.Context, string) (*Instance, error)
-	List(context.Context) ([]*Instance, error)
+	Create(context.Context, *v1alpha1.ECSNodeClass, *karpv1.NodeClaim,
+		[]*cloudprovider.InstanceType) (*ecsclient.DescribeInstancesResponseBodyInstancesInstance, error)
+	Get(context.Context, string) (*ecsclient.DescribeInstancesResponseBodyInstancesInstance, error)
+	List(context.Context) ([]*ecsclient.DescribeInstancesResponseBodyInstancesInstance, error)
 	Delete(context.Context, string) error
 	CreateTags(context.Context, string, map[string]string) error
 }
 
 type DefaultProvider struct {
-	region string
+	ecsClient *ecsclient.Client
+	region    string
 }
 
-func NewDefaultProvider(ctx context.Context, region string) *DefaultProvider {
+func NewDefaultProvider(ctx context.Context, region string, ecsClient *ecsclient.Client) *DefaultProvider {
 	return &DefaultProvider{
-		region: region,
+		ecsClient: ecsClient,
+		region:    region,
 	}
 }
 
-func (p *DefaultProvider) Create(ctx context.Context, nodeClass *v1alpha1.ECSNodeClass, nodeClaim *karpv1.NodeClaim, instanceTypes []*cloudprovider.InstanceType) (*Instance, error) {
+func (p *DefaultProvider) Create(ctx context.Context, nodeClass *v1alpha1.ECSNodeClass, nodeClaim *karpv1.NodeClaim,
+	instanceTypes []*cloudprovider.InstanceType) (*ecsclient.DescribeInstancesResponseBodyInstancesInstance, error) {
 
 	// TODO: implement me
 	return nil, nil
 }
 
-func (p *DefaultProvider) Get(ctx context.Context, id string) (*Instance, error) {
+func (p *DefaultProvider) Get(ctx context.Context, id string) (*ecsclient.DescribeInstancesResponseBodyInstancesInstance, error) {
+	describeInstancesRequest := &ecsclient.DescribeInstancesRequest{
+		RegionId:    &p.region,
+		InstanceIds: tea.String(id),
```

Comment:
I suspect this may be in the wrong format. When called on the console, it has the format requirement. It’s best to verify if possible.
![image](https://github.com/user-attachments/assets/c50719b7-8958-44c6-b7a9-4aeb17880e04)

Reply:
It looks like `["i-bp67acfmxazb4p****"]` is fine, I need to do some adaptations